lib/puppet/resource/type_collection.rb

```diff
@@ -187,7 +187,7 @@ def version
       if environment.config_version.nil? || environment.config_version == ""
         @version = Time.now.to_i
       else
-        @version = Puppet::Util::Execution.execute([environment.config_version]).strip
+        @version = Puppet::Util::Execution.execute(environment.config_version).strip
```

Comment:
Sorry, I mean if you pass an `Array` to ruby's `Kernel.exec` it will execute the process using `execv` directly, bypassing the shell. If you pass a string, it will shell out: https://github.com/ruby/ruby/blob/ruby_2_1/process.c#L3127-L3135.

If we don't arrayify and you're using webrick/passenger, I think we'll end up shelling out when previously we didn't (but it would be good to confirm that).